core/app/models/spree/ability.rb

```diff
@@ -32,44 +45,54 @@ def initialize(user)
       alias_action :new_action, to: :create
       alias_action :show, to: :read
       alias_action :index, :read, to: :display
+    end
 
-
-      user ||= Spree.user_class.new
-
+    def grant_default_permissions
+      # if the user is a "super user" give them full permissions, otherwise give them the permissions
+      # required to checkout and use the frontend.
       if user.respond_to?(:has_spree_role?) && user.has_spree_role?('admin')
         can :manage, :all
       else
-        can :display, Country
-        can :display, OptionType
-        can :display, OptionValue
-        can :create, Order
-        can [:read, :update], Order do |order, token|
-          order.user == user || order.guest_token && token == order.guest_token
-        end
-        can :create, ReturnAuthorization do |return_authorization|
-          return_authorization.order.user == user
-        end
-        can :display, CreditCard, user_id: user.id
-        can :display, Product
-        can :display, ProductProperty
-        can :display, Property
-        can :create, Spree.user_class
-        can [:read, :update, :destroy], Spree.user_class, id: user.id
-        can :display, State
-        can :display, StockItem, stock_location: { active: true }
-        can :display, StockLocation, active: true
-        can :display, Taxon
-        can :display, Taxonomy
-        can [:display, :view_out_of_stock], Variant
-        can :display, Zone
+        grant_generic_user_permissions
       end
+    end
 
-      # Include any abilities registered by extensions, etc.
+    def grant_generic_user_permissions
+      can :display, Country
+      can :display, OptionType
+      can :display, OptionValue
+      can :create, Order
+      can [:read, :update], Order do |order, token|
+        order.user == user || order.guest_token && token == order.guest_token
+      end
+      can :create, ReturnAuthorization do |return_authorization|
+        return_authorization.order.user == user
+      end
+      can :display, CreditCard, user_id: user.id
+      can :display, Product
+      can :display, ProductProperty
+      can :display, Property
+      can :create, Spree.user_class
+      can [:read, :update, :destroy], Spree.user_class, id: user.id
+      can :display, State
+      can :display, StockItem, stock_location: { active: true }
+      can :display, StockLocation, active: true
```

Comment:
yea i guess you're right. i was thinking of a scenario whereby we might want to grant a customer early access to certain items or something along those lines. it could be achieved by granting them access to a special stock location.